src/main/java/org/folio/persist/InstanceRepository.java

```diff
@@ -35,11 +36,69 @@ public class InstanceRepository extends AbstractRepository<Instance> {
   private static final String INSTANCE_SET_VIEW = "instance_set";
   private static final String INSTANCE_HOLDINGS_ITEM_VIEW = "instance_holdings_item_view";
   private static final String INVENTORY_VIEW_JSONB_FIELD = "inventory_view.jsonb";
+  private static final String INSTANCE_SUBJECT_SOURCE_TABLE = "instance_subject_source";
+  private static final String INSTANCE_SUBJECT_TYPE_TABLE = "instance_subject_type";
+
 
   public InstanceRepository(Context context, Map<String, String> okapiHeaders) {
     super(postgresClient(context, okapiHeaders), INSTANCE_TABLE, Instance.class);
   }
 
+  public void linkInstanceWithSubjectSourceAndType(Instance instance) {
+    instance.getSubjects().forEach(subject -> {
+      var sql = new StringBuilder();
+      if (subject.getSourceId() != null) {
+        sql.append(" INSERT INTO ");
+        sql.append(postgresClientFuturized.getFullTableName(INSTANCE_SUBJECT_SOURCE_TABLE));
+        sql.append(" (instance_id, source_id) ");
+        sql.append(" VALUES ");
+        sql.append(String.format("( '%s' , '%s' ) ON CONFLICT DO NOTHING; ", instance.getId(), subject.getSourceId()));
+      }
+      if (subject.getTypeId() != null) {
+        sql.append(" INSERT INTO ");
+        sql.append(postgresClientFuturized.getFullTableName(INSTANCE_SUBJECT_TYPE_TABLE));
+        sql.append(" (instance_id, type_id) ");
+        sql.append(" VALUES ");
+        sql.append(String.format("( '%s' , '%s' ) ON CONFLICT DO NOTHING; ", instance.getId(), subject.getTypeId()));
+      }
+      var sqlString = sql.toString();
+      if (!isBlank(sqlString)) {
+        postgresClient.execute(sql.toString());
+      }
+    });
+  }
+
+  public void unlinkInstanceFromSubjectSource(String instanceId) {
+    var sql = unlinkInstanceFromSubjectSql(INSTANCE_SUBJECT_SOURCE_TABLE, instanceId);
+    postgresClient.execute(sql);
+  }
+
+  public void unlinkInstanceFromSubjectType(String instanceId) {
+    var sql = unlinkInstanceFromSubjectSql(INSTANCE_SUBJECT_TYPE_TABLE, instanceId);
+    postgresClient.execute(sql);
+  }
+
+  private String unlinkInstanceFromSubjectSql(String table, String id) {
+    return String.format("DELETE FROM %s WHERE instance_id = '%s'; ", postgresClientFuturized.getFullTableName(table), id);
+  }
+
+  public void unlinkSubjectSource(String instanceId, String sourceId) {
+    var sql = "DELETE FROM " +
+      postgresClientFuturized.getFullTableName(INSTANCE_SUBJECT_SOURCE_TABLE) +
+      String.format(" WHERE instance_id = '%s' AND source_id = '%s';",
+        instanceId, sourceId);
+    postgresClient.execute(sql);
+  }
+
+  public void unlinkSubjectType(String instanceId, String typeId) {
+    var sql = "DELETE FROM " +
+      postgresClientFuturized.getFullTableName(INSTANCE_SUBJECT_TYPE_TABLE) +
+      String.format(" WHERE instance_id = '%s' AND type_id = '%s';", instanceId,
+        typeId);
+    postgresClient.execute(sql);
```

Comment:
Basicaly the logic should be like following:
1. Get all source/type ids from new instance subjects.
2. Get all source/type ids from old instance subjects.
3. Identify distinction: what should be created, what should be deleted.
4. Do queries for creating and deleting these relations in batch and not one by one.